### PR TITLE
[WIP] KEP: introduce GPUTopologyAware plugin

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Xref https://github.com/kubernetes-sigs/scheduler-plugins/issues/690

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by kerthcet** [comment]
/assign @Huang-Wei 
cc @lengrongfu

File: kep/690-gpu-topology-aware-scheduling/README.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+    maxScore = limitQuota * (limitQuota - 1) * 1800 / 2 // 1800 is the maximum score between two GPUs, on behalf of 18 NVLinks
+    finalScore = originalScore * 100 / maxScore
+    ```
+
+A potential issue here is if the score of the link type is too small, like `P2PLinkCrossCPU` maps to 10, then we may loss the precision here
+and got 0 score finally.
+
+**In summary**
+
+Based on the same scoring algorithm, kube-scheduler will find the best-fit node, but no need to tell the device plugin which GPUs are they,
```
Comment (lianghao208) [inline kep/690-gpu-topology-aware-scheduling/README.md:282]: IIIUC, kube-scheduler reuses NVIDIA/go-gpuallocator algorithm to calculate the best GPU device id for allocation and score the nodes accordingly. But scheduler can't 100% sure which GPU will be used eventually, it "guesses" the result depends on the topology information reported by NFD, and device plugin will make its own decision, am I right? 
If so, does scheduler get to know which GPU has already been used?

Comment (kerthcet) [reply kep/690-gpu-topology-aware-scheduling/README.md:282]: Missed that part, will add it. Thanks.

**Comment by KunWuLuan** [comment]
Scheduler may allocate pod in different sequence with device plugin, will this cause problem? For example, scheduler may allocate 4 GPUs for pod A and allocate 2 GPUs for another pod, but kubelet may handle pod with 2 GPUs first.

**Comment by KunWuLuan** [comment]
Maybe we can implement a new plugin to ensure scheduler make same decision with kubelet, this new plugin can be a common plugin and can be used by many other plugins, like gpu topology or rdma topology. Then gpu topology just need to do its own decisions.

**Pull request closed without merging**